test/modules/post/test/cmd_exec.rb

```diff
@@ -111,4 +111,60 @@ def test_cmd_exec_stderr
       end
     end
   end
+
+  def test_cmd_exec_channelized
+    vprint_status("Starting cmd_exec channelized tests")
+
+    it "should run channelized command and wait for output with an extra channel" do
+      test_string = Rex::Text.rand_text_alpha(4)
+      timeout = 15
+      channel_count = session.channels.count
+
+      if session.platform.eql? 'windows'
+        output = cmd_exec("cmd.exe", "/c echo #{test_string}", timeout, { 'Channelized' => true})
+        output.rstrip == test_string && session.channels.count == channel_count + 1
+      else
+        output = cmd_exec("echo #{test_string}", nil, timeout, { 'Channelized' => true})
+        output == test_string && session.channels.count == channel_count + 1
+      end
+    end
+
+    it "should run non-channelized command and not wait for output without an extra channel" do
+      test_string = Rex::Text.rand_text_alpha(4)
+      timeout = 15
+      channel_count = session.channels.count
+
+      if session.platform.eql? 'windows'
+        cmd_exec("cmd.exe", "/c echo #{test_string}", timeout, { 'Channelized' => false})
+      else
+        cmd_exec("echo #{test_string}", nil, timeout, { 'Channelized' => false})
```

Comment:
Can we throw in a quick assertion that this returns nil, or an empty string, or whatever is expected 📈